src-tauri/src/tray_icon_gen.rs

```diff
@@ -0,0 +1,74 @@
+use tauri::{AppHandle, Manager};
+use once_cell::sync::Lazy;
+use std::io::Cursor;
+use ril::prelude::*;
+use ril::text::{Font, TextLayout, VerticalAnchor, TextSegment};
+
+static FONT: Lazy<Font> = Lazy::new(|| {
+    let font_data = include_bytes!("../fonts/SFMonoRegular.otf");
+    Font::from_bytes(font_data, 16.0).expect("Failed to load embedded font")
+});
+
+#[tauri::command]
+pub fn generate_timer_icon(
+    app_handle: AppHandle,
+    time_string: String,
+    current_ms: f64,
+    total_ms: f64,
+) -> Result<Vec<u8>, String> {
+    let scale_factor = app_handle
+        .get_webview_window("main")
+        .map(|w| w.scale_factor().unwrap_or(1.0))
+        .unwrap_or(1.0);
+
+    let font: &Font = &*FONT;
+
+    let base_width = 88.0;
+    let base_height = 22.0;
+    let base_font_size = 16.0;
+    let base_padding = 4.0;
+
+    let width = (base_width * scale_factor as f64).round() as u32;
+    let height = (base_height * scale_factor as f64).round() as u32;
+    let font_size = (base_font_size * scale_factor as f64) as f32;
+    let padding = (base_padding * scale_factor as f64).round() as u32;
+    let progress_bar_color = Rgba::new(87u8, 32u8, 174u8, 220u8);
+
+    let text_color = Rgba::new(255u8, 255u8, 255u8, 255u8);
+
+    let box_color = Rgba::new(3u8, 7u8, 21u8, 180u8);
+
+    let mut img = Image::new(width, height, Rgba::new(0, 0, 0, 0));
+
+    let percentage = if total_ms > 0.0 { (current_ms / total_ms).min(1.0).max(0.0) } else { 0.0 };
```

Comment:
I'd recommend splitting out the calculation for the green bar width into its own function `get_green_bar_width`